.github/workflows/format-and-lint.yaml

```diff
@@ -0,0 +1,34 @@
+on:
+  workflow_call:
+
+jobs:
+  format-and-lint:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v5
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      # Change ruffConfig to main before merge
+      - name: Install dependencies
+        run: |
+          wget https://raw.githubusercontent.com/mdolab/.github/ruffConfig/.pre-commit-config.yaml
+          pip install pre-commit
+          # If there's already a ruff.toml file in the repo, we should rename it to ruff.toml-project, then download the mdolab ruff.toml file edit it to put the line `extend = "ruff.toml-project"` at the top so that ruff will use both configurations.
+          if [[ -f "ruff.toml" ]]; then
+            mv ruff.toml ruff.toml-project;
+          fi
+          wget https://raw.githubusercontent.com/mdolab/.github/ruffConfig/ruff.toml
+          if [[ -f "ruff.toml-project" ]]; then
+              sed -i '1s/^/extend = "ruff.toml-project" \n/' ruff.toml
+          fi
```

Comment:
Maybe this works both ways, but should this not be the other way around? Based on the [docs](https://docs.astral.sh/ruff/settings/#extend) it seems to add this line to the project config file extending the base config file. The project file should then be passed to ruff.
I have not tested this, but it seems that this can cause some confusion or bugs, since the expectation is that the project file is more restrictive than the global config. This offers the opposite, the base config will overwrite the project config. Am I missing something?